specs/agents/tracing-api-otel.md

```diff
@@ -0,0 +1,128 @@
+## OpenTelemetry API (Tracing)
+
+[OpenTelemetry](https://opentelemetry.io) (OTel in short) provides a vendor-neutral API that allows to capture tracing, logs and metrics data.
+
+Agents may provide a bridge implementation of OpenTelemetry Tracing API following this specification.
+
+Bridging here means that for each OTel span created with the API, a native span/transaction will be created and sent to APM server.
+
+From the perspective of the application code calling the OTel API, the delegation to a native span/transaction is transparent.
+Also, this means that any OTel processors will be bypassed and ignored by the bridge.
+
+### Spans and Transactions
+
+OTel only defines Spans, whereas Elastic APM relies on both Spans and Transactions.
+OTel allows users to provide the _remote context_ when creating a span, which is equivalent to providing a parent to a transaction or span,
+it also allows to provide a (local) parent span.
+
+As a result, when creating Spans through OTel API with a bridge, agents must implement the following algorithm:
+
+```javascript
+// otel_span contains the properties set through the OTel API
+span_or_transaction = null;
+if (otel_span.remote_contex != null) {
+    span_or_transaction = createTransactionWithParent(otel_span.remote_context);
+} else if (otel_span.parent == null) {
+    span_or_transaction = createRootTransaction();
+} else {
+    span_or_transaction = createSpanWithParent(otel_span.parent);
+}
+```
+
+### Span Kind
+
+OTel spans have an `SpanKind` property ([specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind)) which is close but not strictly equivalent to our definition of spans and transactions.
+
+For both transactions and spans, an optional `otel.span_kind` property will be provided by agents when set through
+the OTel API.
+This value should be stored into Elasticsearch documents to preserve OTel semantics and help future OTel integration.
+
+Possible values are `CLIENT`, `SERVER`, `PRODUCER`, `CONSUMER` and `INTERNAL`, refer to [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind) for details on semantics.
+
+When `otel.span_kind` is not provided by the agent, APM Server should infer it using the following algorithm:
+
+```javascript
+span_kind = null;
+if (isTransaction(item)) {
+    if (item.type == "messaging") {
+        span_kind = "CONSUMER";
+    } else if (item.type == "request") {
+        span_kind = "SERVER";
+    }
+} else {
+    // span
+    if (item.type == "external" || item.type == "storage") {
```

Comment:
Should `item.type == "db"` also be included in this?

Reply:
Thanks for catching this ! I've just updated the spec.